### PR TITLE
PersonNote and EntityNote support

### PR DESCRIPTION
**Pull request by guymaness2**
##### What does the change add or fix?

Fixed a bug with legacy Note import affecting notes of entity type other than Person. Also added new, optimized support for importing PersonNotes and Entity Notes.

---------

##### What does the change add or fix in a succinct statement that will be read by clients?

* Fixed bug in Legacy Note import logic causing errors importing notes with entity type other than Person.
* Added new V2 handling of PersonNotes.
* Added new V2 handling of EntityNotes.

---------

##### Who reported, requested, or paid for the change?

Internal

---------

##### Does this update or add options to the block UI?

No

---------

##### What files does it affect?

* Bulldozer.CSV/CSVComponent.cs
* Bulldozer.CSV/CSVInstance.cs
* Bulldozer.CSV/Maps/Notes.cs
* Bulldozer.CSV/Model/EntityNoteCsv.cs
* Bulldozer/Model/NoteImport.cs. 

---------

##### Is it a breaking change for other versions/clients?

No

### CODE REVIEW

File: Bulldozer.CSV/SampleCSVs/person-notes.csv [hunk 1]
```diff
@@ -0,0 +1,3 @@
+﻿Id,NoteType,Caption,IsAlert,IsPrivateNote,NotificationSent,Text,EntityTypeName,PersonId,DateTime,CreatedByPersonId
+35746,General,Barbara - Care Room,0,0,0,Struggling with empathy for other people,Rock.Model.Person,140,22:20.5,
```
Comment (guymaness2) [inline Bulldozer.CSV/SampleCSVs/person-notes.csv:2]: Fix date

**Comment by guymaness2** [comment]
Need to add entity-note csv file

**Review by nateh777 — CHANGES_REQUESTED** [review]
See Guy's notes

**Comment by guymaness2** [comment]
@nateh777 Check out the final 3 commits. Should be good to go now. There was one slight tweak to the bulldozer code for attendance, but I think it should make sense to you. Basically, attendance csv provides a row for every attendance input, but also carries the information needed to create the occurrences. In implementing support for Occurrence.Notes fields, the logic needed tightened up to concatenate any notes that may appear across the attendance values. Before it was just pulling the first row it came across for the occurrence, ignoring any notes in any others rows for the same occurrence.

File: Bulldozer.CSV/SampleCSVs/entity-notes.csv [hunk 2]
```diff
@@ -0,0 +1,2 @@
+﻿Id,NoteType,Caption,IsAlert,IsPrivateNote,NotificationSent,Text,EntityTypeName,EntityId,DateTime,CreatedByPersonId
+4183,Group Member Note,,0,0,0,Visited with Sara Meese and asked to join the group right away.,Rock.Model.GroupMember,112_147,2021-03-25 17:22:20.473,
```
Comment (nateh777) [inline Bulldozer.CSV/SampleCSVs/entity-notes.csv:2]: Probably just nitpicky, but does it matter the date format? Previous csv's https://github.com/KingdomFirst/Bulldozer/blob/master/Bulldozer.CSV/SampleCSVs/person.csv?plain=1 and https://github.com/KingdomFirst/Bulldozer/blob/master/Bulldozer.CSV/SampleCSVs/attendance.csv?plain=1 have American date format.
```suggestion
4183,Group Member Note,,0,0,0,Visited with Sara Meese and asked to join the group right away.,Rock.Model.GroupMember,112_147,03/25/2021 17:22:20.473,
```

File: Bulldozer.CSV/SampleCSVs/person-notes.csv [hunk 3]
```diff
@@ -1,3 +1,3 @@
-﻿Id,NoteType,Caption,IsAlert,IsPrivateNote,NotificationSent,Text,EntityTypeName,PersonId,DateTime,CreatedByPersonId
-35746,General,Barbara - Care Room,0,0,0,Struggling with empathy for other people,Rock.Model.Person,140,22:20.5,
-35747,Volunteer Process,Security Concerns,1,0,0,Should serve in visible ministry areas only.,Rock.Model.Person,145,22:20.5,
+﻿Id,NoteType,Caption,IsAlert,IsPrivateNote,NotificationSent,Text,PersonId,DateTime,CreatedByPersonId
+35746,General,Barbara - Care Room,0,0,0,Struggling with empathy for other people,140,2014-01-10 07:33:12.031,
+35747,Volunteer Process,Security Concerns,1,0,0,Should serve in visible ministry areas only.,2016-05-16 17:22:20.473,
```
Comment (nateh777) [inline Bulldozer.CSV/SampleCSVs/person-notes.csv:3]:
```suggestion
35746,General,Barbara - Care Room,0,0,0,Struggling with empathy for other people,140,01/10/2014 07:33:12.031,
35747,Volunteer Process,Security Concerns,1,0,0,Should serve in visible ministry areas only.,05/16/2016 17:22:20.473,
```

**Review by nateh777 — CHANGES_REQUESTED** [review]
@guymaness2 Minor suggestions, since I noticed it in your attendance.csv file, otherwise probably wouldn't have noticed it.

**Review by nateh777 — APPROVED** [review]

**Pull request merged**